app/models/book.rb

```diff
@@ -19,6 +17,10 @@ def to_s
     slug
   end
 
+  def discussions_in_organization(organization = Organization.current)
+    Discussion.all.where(organization: organization).includes(exercise: [:language, :guide])
```

Comment:
Getting all the discussions for a book this way instead of getting all the exercises and getting all the discussions through there it's a **huge** performance improvement. Also, pre-fetching language and guide improves things too.

Reply:
`Discussion.all.where` == `Discussion.where`?